resources/saas-boost-idp.yaml

```diff
@@ -253,30 +253,54 @@ Resources:
         StartCodeBuildLambda: !Ref StartCodeBuildLambda
         CloudFormationWaitHandleCallback: !Ref CloudFormationWaitHandleCallback
         AdminCredentials: !Ref SaaSBoostAdminCredentials
+        AdminWebUrl: !Ref AdminWebUrl
         VPC: !Ref VPC
         PublicSubnets: !Ref PublicSubnets
         PrivateSubnets: !Ref PrivateSubnets
-Outputs:
-  SBUserPool:
-    Condition: UseCognito
-    Description: SaaS Boost User Pool Id
-    Value: !Ref UserPool
-  SBUserPoolClient:
-    Condition: UseCognito
-    Description: SaaS Boost User Pool Client id
-    Value: !Ref UserPoolClient
-  CognitoSignOnBaseUri:
-    Condition: UseCognito
-    Description: URL for Cognito Hosted signon base URI
-    Value: !Sub saas-boost-${Environment}.auth.${AWS::Region}.amazoncognito.com
-  KeycloakHost:
+  AdminWebAppClientName:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: !Sub /saas-boost/${Environment}/ADMIN_WEB_APP_CLIENT
+      Type: String
+      Value: !If [UseCognito, !Ref UserPoolClient, !GetAtt keycloak.Outputs.AdminWebAppClientName]
+  AdminWebAppClientId:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: !Sub /saas-boost/${Environment}/ADMIN_WEB_APP_CLIENT_ID
+      Type: String
+      Value: !If [UseCognito, !Ref UserPoolClient, !GetAtt keycloak.Outputs.AdminWebAppClientId]
+  ApiAppClientId:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: !Sub /saas-boost/${Environment}/API_APP_CLIENT_ID
+      Type: String
+      Value: !If [UseCognito, !Ref UserPoolClient, !GetAtt keycloak.Outputs.ApiAppClientId]
+  ApiAppClient:
+    Type: AWS::SecretsManager::Secret
     Condition: UseKeycloak
-    Description: URL for the Keycloak install
-    Value: !GetAtt keycloak.Outputs.KeycloakHost
-  KeycloakRealm:
+    Properties:
+      Name: !Sub sb-${Environment}-api-app-client
+      SecretString: !Sub '{"client_name": "${keycloak.Outputs.ApiAppClientName}", "client_id": "${keycloak.Outputs.ApiAppClienId}", "client_secret": "${keycloak.Outputs.ApiAppClientSecret}"'
+Outputs:
+  OidcHost:
+    Description: URL of the IdP for hosted sign in
+    Value: !If
+      - UseCognito
+      - !Sub saas-boost-${Environment}.auth.${AWS::Region}.amazoncognito.com
+      - !GetAtt keycloak.Outputs.KeycloakHost
+  OidcIssuer:
+    Description: OIDC issuer for System User IdP
+    Value: !If
+      - UseCognito
+      - !Sub saas-boost-${Environment}.auth.${AWS::Region}.amazoncognito.com/${UserPool}
+      - !GetAtt keycloak.Outputs.KeycloakIssuer
```

Comment:
I've only tested with `https://cognito-idp.${AWS::Region}.amazonaws.com/${UserPool}`, have you verified this works the same way?